tests/scalers/azure-service-bus-queue-workload-identity.test.ts

```diff
@@ -0,0 +1,211 @@
+import * as sh from "shelljs"
+import * as azure from "@azure/service-bus"
+import test from "ava"
+import { createNamespace, createYamlFile, waitForDeploymentReplicaCount } from "./helpers"
+
+const connectionString = process.env["AZURE_SERVICE_BUS_CONNECTION_STRING"]
```

Comment:
IDK how workload identity works, don't we need to create any binding with anything in order to use the msi?

Reply:
Unlike pod identity which uses CRDs such as `AzureIdentity` and `AzureIdentityBinding`, workload identity only relies on native kubernetes resources such as the service account and deployments.


As long as the service account used by the KEDA pods has the [required](https://azure.github.io/azure-workload-identity/docs/topics/service-account-labels-and-annotations.html) labels and annotations, the mutating webhook component that is setup for workload identity will insert the environment variables into the pods using the service account.

These variables are then used in the code to fetch the token .

Reply:
We just have to setup the mutating webhook on the test cluster, and the service account is configured correctly (can be done using the new helm flags introduced).

As long as these are done, the test will work.

Reply:
Since you have access to the cluster, are you up for setting it up?

Reply:
I can try to set it up. Had a query. When installing KEDA via `helm` we can use CLI flags to configure the values that are used in the configuration. In our testing environment, we setup KEDA using `make deploy`, is there a way to use these flags?

Reply:
Don't have the necessary permissions.

Reply:
AFAIK, it's not possible. Maybe you could add some logic there to add the needed label using kustomize in case of running e2e tests but right now it's not done

Reply:
I am setting the require annotations in the `setup.test.ts` file.